docs/general.md

```diff
@@ -534,4 +584,82 @@ add 8,r0,r1
 
 ## Bootloader
 
-Operator execution (transactions come in, get executed on the bootloader, state is suspended until new transaction shows up).
+The **Bootloader** is a system contract that orchestrates the construction and validation of new blocks, acting as a bridge between the EraVM and the external server to ensure accurate transaction processing.
+
+It validates and executes transactions, handles errors effectively, and integrates with EraVM for seamless block construction.
+
+### Core Functions
+1. **Transaction Lifecycle Management**:
+    - **Validation**: Validates each transaction from the server to meet the required criteria.
+    - **Fee Processing**: Calculates and deducts the necessary transaction fees.
+    - **Execution**: Executes transactions, forming the new block.
+
+    Simplified workflow:
+    ```solidity
+    contract Bootloader {
+        function executeBlock(address operatorAddress, Transaction[2] memory transactions) {
+            for (uint256 i = 0; i < transactions.length; i++) {
+                validateTransaction(transactions[i]);
+                chargeFee(operatorAddress, transactions[i]);
+                executeTransaction(transactions[i]);
+            }
+        }
+    }
+    ```
+2. **Error Handling**:
+    - **Malformed Transactions**: Reverts EraVM to the last valid state if a transaction is malformed.
+    - **Contract Errors**: Rolls back to a safe checkpoint on contract errors (e.g., revert, panic) to maintain system integrity.
+3. **Integration with EraVM**:
+    - Retrieves operational code from the `Decommitter` using a hash from the server.
+    - Uses its heap to dynamically interface with the server and process transaction data systematically.
+4. **Contract Code Management**: Retrieves or defaults contract code based on specific conditions, ensuring correct handling during contract calls.
+
+#### Bootloader Interaction with `decommit_code_hash`
+
+The `decommit_code_hash` function is essential for how the Bootloader handles contracts and their associated code within the EraVM system.
+
+1. **Contract Identification**: The Bootloader identifies the system's deployer contract address, generating a `storage_key` to access the contract's code in storage:
+    ```rust
+    let deployer_system_contract_address = Address::from_low_u64_be(DEPLOYER_SYSTEM_CONTRACT_ADDRESS_LOW as u64);
+    let storage_key = StorageKey::new(deployer_system_contract_address, address_into_u256(address));
+    ```
```

Comment:
These sections (until 5) are a bit odd, if I dind't know the bootloader I would think it is written in Rust, maybe specify a little more that that code is from the EraVM `decommit_code_hash` and how does the Bootloader access them

Reply:
Just before it says it is from EraVM. Should I add anything else?
> The decommit_code_hash function is essential for how the Bootloader handles contracts and their associated code within the EraVM system.
When you said _how does the Bootloader access them_, do you mean the contracts or the hash of the code generated by this function?

Reply:
What I mean is since it says 
`The Bootloader identifies the system's deployer contract address, generating a storage_key to access the contract's code in storage:`
It seems like the bootloader is doing that in Rust, when its the function `decommit_code_hash` that is doing that
With _how does the Bootloader access them_ i meant how does it access the `decommit_code_hash` function

Reply:
you are right this is wrong